generator/altitude_generator.cpp

```diff
@@ -0,0 +1,86 @@
+#include "generator/routing_generator.hpp"
+#include "generator/srtm_parser.hpp"
+
+#include "defines.hpp"
+
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/feature.hpp"
+#include "indexer/feature_altitude.hpp"
+#include "indexer/feature_data.hpp"
+#include "indexer/feature_processor.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/varint.hpp"
+
+#include "coding/internal/file_data.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/map.hpp"
+#include "std/type_traits.hpp"
+
+using namespace feature;
+
+namespace
+{
+static_assert(is_same<TAltitude, generator::SrtmTile::THeight>::value, "");
+static_assert(kInvalidAltitude == generator::SrtmTile::kInvalidHeight, "");
+
+class Processor
+{
+public:
+  Processor(string const & srtmPath) : m_srtmManager(srtmPath) {}
+  map<uint32_t, Altitudes> const & GetFeatureAltitudes() const { return m_featureAltitudes; }
+
+  void operator()(FeatureType const & f, uint32_t const & id)
+  {
+    f.ParseTypes();
+    f.ParseHeader2();
```

Comment:
1. Without 
```c++
f.ParseTypes();
f.ParseHeader2();
```
everything works fine and I removed these lines of code.

2. After I removed these line the size of "altitude" section in resulting mwm wasn't changed. But the size of whole mwm was changed. It's because of osrm section. Do you know why it could happen? You can see below two mwm dumps.
```
bash-3.2$ ../omim/tools/python/mwm/dump_mwm.py ./Russia_Astrakhan\ Oblast.mwm
Tags:
  version : offs         8 len       10
  header  : offs        24 len       32
  dat     : offs        56 len  1820993
  geom0   : offs   1821056 len    25614
  trg0    : offs   1846672 len    52068
  geom1   : offs   1898744 len    98842
  trg1    : offs   1997592 len   191329
  geom2   : offs   2188928 len   362368
  trg2    : offs   2551296 len   440628
  geom3   : offs   2991928 len   928086
  trg3    : offs   3920016 len   779501
  metaidx : offs   4699520 len    22320
  meta    : offs   4721840 len    69451
  offs    : offs   4791296 len    81148
  idx     : offs   4872448 len   396947
  sdx     : offs   5269400 len   387680
  addr    : offs   5657080 len    32539
  ranks   : offs   5689624 len    17096
  altitude: offs   5706720 len   193072
  node2ftseg: offs   5899792 len     8412
  ftseg   : offs   5908208 len   431184
  skoda   : offs   6339392 len    58208
  daewoo  : offs   6397600 len   625020
  mercedes: offs   7022624 len   811456
  infinity: offs   7834080 len   392048
Format: 8, version: 2016-06-27 11:05
Header: {'langs': [], 'mapType': 'country', 'basePoint': (46.8896554923777, 47.3665681557535), 'bounds': (44.26537949044541, 39.77490682039603, 53.028132267505924, 49.27907749436055), 'scales': [10, 12, 14, 17]}
Metadata count: 2790
Sample features:
{"geometry": {"type": "Point", "coordinates": [47.689002064533554, 47.170044280034524]}, "header": {"types": ["power|tower"]}, "size": 8}
{"geometry": {"type": "Polygon"}, "header": {"house": "38", "types": ["building"]}, "size": 16}
{"geometry": {"type": "LineString"}, "header": {"name": {"default": "Школа одаренных детей"}, "types": ["barrier|fence"]}, "size": 73}
{"geometry": {"type": "LineString"}, "header": {"types": ["railway|rail"]}, "size": 21}
{"geometry": {"type": "Polygon"}, "header": {"types": ["building"]}, "size": 14}


bash-3.2$ ../src_github/omim/tools/python/mwm/dump_mwm.py ./Russia_Astrakhan\ Oblast.mwm
Tags:
  version : offs         8 len       10
  header  : offs        24 len       32
  dat     : offs        56 len  1820993
  geom0   : offs   1821056 len    25614
  trg0    : offs   1846672 len    52068
  geom1   : offs   1898744 len    98842
  trg1    : offs   1997592 len   191329
  geom2   : offs   2188928 len   362368
  trg2    : offs   2551296 len   440628
  geom3   : offs   2991928 len   928086
  trg3    : offs   3920016 len   779501
  metaidx : offs   4699520 len    22320
  meta    : offs   4721840 len    69451
  offs    : offs   4791296 len    81148
  idx     : offs   4872448 len   396947
  sdx     : offs   5269400 len   387680
  addr    : offs   5657080 len    32539
  ranks   : offs   5689624 len    17096
  altitude: offs   5706720 len   193072
  node2ftseg: offs   5899792 len     8424
  ftseg   : offs   5908216 len   431192
  skoda   : offs   6339408 len    58216
  daewoo  : offs   6397624 len   624960
  mercedes: offs   7022584 len   811484
  infinity: offs   7834072 len   392216
Format: 8, version: 2016-06-27 10:10
Header: {'langs': [], 'mapType': 'country', 'basePoint': (46.8896554923777, 47.3665681557535), 'bounds': (44.26537949044541, 39.77490682039603, 53.028132267505924, 49.27907749436055), 'scales': [10, 12, 14, 17]}
Metadata count: 2790
Sample features:
{"geometry": {"type": "Polygon"}, "header": {"types": ["building"]}, "size": 15}
{"geometry": {"type": "LineString"}, "header": {"types": ["waterway|stream"]}, "size": 18}
{"geometry": {"type": "Polygon"}, "header": {"types": ["building"]}, "size": 15}
{"geometry": {"type": "LineString"}, "header": {"name": {"default": "Садовая улица"}, "types": ["highway|residential"]}, "size": 51}
{"geometry": {"type": "Polygon"}, "header": {"types": ["natural|wetland"]}, "size": 12}
```